lisa/nic.py

```diff
@@ -147,26 +176,41 @@ def __len__(self) -> int:
         return len(self.nics)
 
     def append(self, next_node: NicInfo) -> None:
-        self.nics[next_node.upper] = next_node
+        self.nics[next_node.name] = next_node
 
     def is_empty(self) -> bool:
         return len(self.nics) == 0
 
     def get_unpaired_devices(self) -> List[str]:
-        return [x.upper for x in self.nics.values() if not x.lower]
+        return [x.name for x in self.nics.values() if not x.pci_device_name]
```

Comment:
"unpaired" means it has no upper and no lower.

Reply:
current logic is to try to find the pairs firstly, then remain ones will be initialized as upper nics, so there is no cases for it has lower but no upper.

Reply:
how about below? It looks clearer.
`[x.name for x in self.nics.values() if not x.lower]`

Reply:
there is no `lower` now, I use pci_device_name instead.

Reply:
The lower should be kept to support all scenarios. For example, how to differentiate single functional pci device (like bare metal), synthetic device with slave pci? What's difference between systhetic/pci pair?